### PR TITLE
dev/core#1731 - Unable to import relationship

### PR DESCRIPTION
**Pull request by jitendrapurohit**
Overview
----------------------------------------
Import returns an error if relationship columns are included.

Before
----------------------------------------
See steps to replicate on gitlab - https://lab.civicrm.org/dev/core/-/issues/1731

image

After
----------------------------------------
Fixed. Import completes without any errors.

Technical Details
----------------------------------------
As there are 2 phone columns, the empty column assigns a null string to the key in `$params` leading to the above error when a value from the next column is assigned in the array format.

Comments
----------------------------------------
Gitlab - https://lab.civicrm.org/dev/core/-/issues/1731

### CODE REVIEW

**Comment by demeritcowboy** [comment]
Hi @jitendrapurohit I've tried looking at this twice before and while the issue itself is clear, I kept getting put off on the r-tech part while trying to look at this function. I gave it another shot today and I'm at about 75% grok level and I see something:

This loop does an iteration over the fields, but keeps a running-total array of the related contact stuff, so what happens now is that if you have yet another phone field AFTER your filled in one and it is blank, it blanks out the whole array, e.g.

FirstName,LastName,Spouse-first-name,spouse-last-name,spouse-home-phone,spouse-work-phone,spouse-home-mobile
Amy,Jones,Doug,Jones,,123-456-7890,

I've tested and it does indeed blank out all the phones. I'm not sure I have a good suggestion - I'm not fully getting this function yet.

There's also a deprecation warning but I expect that was there before - haven't double-checked yet

**Comment by BettyDolfing** [comment]
test this please

**Comment by jaapjansma** [comment]
* General standards
    * Explain ([`r-explain`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-explain))
        * __PASS__ : The goal/problem/solution have been adequately explained in the PR.
    * User impact ([`r-user`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-user))
        * __PASS__: The change would be intuitive or unnoticeable for a majority of users who work with this feature.
    * Documentation ([`r-doc`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-doc))
        * __PASS__: The changes do not require documentation.
    * Run it ([`r-run`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-run))
        * __PASS__: We created a test csv file according to the issue on lab.civicrm.org. We then imported this and set the right mapping. We used Update at For Duplicate Contacts on the first import screen. We did not check the fill option.
* Developer standards
    * Technical impact ([`r-tech`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-tech))
        * __PASS__: The change preserves compatibility with existing callers/code/downstream.
    * Code quality ([`r-code`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-code))
        * __PASS__: The functionality, purpose, and style of the code seems clear+sensible.
    * Maintainability ([`r-maint`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-maint))
        * __PASS__: The change is trivial enough that it does not require tests.
    * Test results ([`r-test`](https://docs.civicrm.org/dev/en/latest/standards/review/#r-test))
        * __PASS__: The test results are all-clear.

@eileenmcnaughton or @mattwire can one of you merge this pr?

**Comment by jitendrapurohit** [comment]
Thanks @jaapjansma & @BettyDolfing for the testing. 

sorry, I've not yet checked the comment from @demeritcowboy above. But I think the issue explained by him makes sense here and could result in a problem if the columns in the csv are in a different order provided in the gitlab. I'll take another look at this tomorrow, replicate and probably include the fix for the additional problem if any.

**Comment by jaapjansma** [comment]
@jitendrapurohit thanks for clarifying we did not fully understand @demeritcowboy comment and we assumed that when the error was gone it was sufficient. So thank @demeritcowboy for testing this thouroughly.

**Comment by demeritcowboy** [comment]
Sorry I might have been a bit too wordy. I was also trying to document what I'd learned about that function. I'm doing it again now aren't I...

Anyway also cross-referencing to note about the deprecation notice - https://github.com/civicrm/civicrm-core/pull/17340. It's not caused by this PR.

**Comment by jitendrapurohit** [comment]
I was able to replicate the behavior mentioned by @demeritcowboy above and have now updated the PR with the fix. Can you confirm if it is working for you now?

Thank you all!

**Comment by demeritcowboy** [comment]
Did some r-runs. Working now.
Thanks.

**Comment by eileenmcnaughton** [comment]
Thanks @jitendrapurohit @demeritcowboy @jaapjansma @BettyDolfing
